### PR TITLE
Ability to define the host interface to bind to.

### PR DESCRIPTION
**Pull request by ibigpapa**
Fixes # 0.

Changes proposed in this pull request:

 - Ability to define the host interface to bind to.

### CODE REVIEW

**Comment by hjacobs** [comment]
Thanks for the PR. This is a backwards-incompatible signature change (for users calling with non-keyword args), probably we can add the `host` parameter at the end instead?

File: connexion/app.py [hunk 1]
```diff
@@ -246,6 +250,11 @@ def run(self, port=None, server=None, debug=None, **options):  # pragma: no cove
             self.port = port
         elif self.port is None:
             self.port = 5000
+        
+        if host is not None:
```
Comment (hjacobs) [inline connexion/app.py]: we could make it shorter: `self.host = host or self.host or '0.0.0.0'`

Comment (ibigpapa) [inline connexion/app.py]: yeah good call on backwards compat.

File: connexion/app.py [hunk 2]
```diff
@@ -265,15 +271,15 @@ def run(self, port=None, server=None, debug=None, **options):  # pragma: no cove
                 raise Exception('tornado library not installed')
             wsgi_container = tornado.wsgi.WSGIContainer(self.app)
             http_server = tornado.httpserver.HTTPServer(wsgi_container, **options)
-            http_server.listen(self.port)
-            logger.info('Listening on port %s..', self.port)
+            http_server.listen(self.port, address=host)
+            logger.info('Listening on port %s.. on interface %s..', self.port, host)
```
Comment (rafaelcaricio) [inline connexion/app.py]: Maybe better be `logger.info('Listening on %s:%s ..', self.host, self.port)`

**Comment by overquota** [comment]
maybe you can change default host to '::' instead of '0.0.0.0'
Reasons: 
> By default listen host='::' interface instead of host='0.0.0.0'. App would think its in ipv6-only world, but you could access it thru ipv4 cause OS would map your ipv4 addr to ipv6 (see http://stackoverflow.com/a/21689979)

And then i would be able to close my pr as duplicate

File: connexion/app.py (hunk 2 above)
Comment (ibigpapa) [inline connexion/app.py]: Agreed

**Comment by ibigpapa** [comment]
Over quota is that behavior is specific? I've not taken the time to review in depth flask run implementation and the under pinning libraries... The next question would be does this work in gevent and tornado?

**Comment by overquota** [comment]
Windows, meh.. you are right, thank you. This is Flask/werkzeug/python http server issue, there are no IPV6_V6ONLY=0 option for socket, wich may be =1 by default in some OSes (Windows for example: https://msdn.microsoft.com/ru-ru/library/windows/desktop/ms738574(v=vs.85).aspx). Some old versions (e.g. XP) not support this option at all :(
okay...jpg

**Comment by rafaelcaricio** [comment]
After this discovery about Windows not supporting the host `::` by default I think we better keep the `0.0.0.0` as default. Let's leave Connexion to _work by default_.

Now with the option of customising the host, after this PR gets merged, users with advanced knowledge can serve their apps over IPv4 and IPv6 sockets setting the host to `::`.

**Comment by rafaelcaricio** [comment]
👍

**Pull request merged**